pkg/generate/app/componentref.go

```diff
@@ -299,6 +302,9 @@ type ComponentInput struct {
 
 	ExpectToBuild bool
 	ScratchImage  bool
+	// For keeping track of if this image was retrieved from a Dockerfile,
+	// which changes our BuildConfig results
```

Comment:
@rhcarvalho in your scenario, FromDockerfile would be false.

FromDockerfile is true when we figured out which image you want to use by parsing the Dockerfile you supplied (either in the repo or via the -D flag).

@oatmealraisin it would be better if this comment explained *how* it changes the results.  "when true, an imagechangetrigger will not be associated with the buildconfig because the user has not expressed a desire to reference an imagestream, they are directly referencing an image"  or some such thing.